Integrations/integration-ProtectWise.yml

```diff
@@ -230,10 +239,19 @@ script:
             return false;
         case 'fetch-incidents':
             var lastRun = getLastRun();
-            var startTime = (lastRun && lastRun.startTime ? lastRun.startTime : (new Date().getTime()- 1000 * 60 ));
-            var data = eventsSearch(startTime , new Date().getTime() , params.eventType, params.killChainStage, params.threatLevel,
+            var now = (new Date()).getTime();
+            if (!lastRun || !lastRun.time) {
+                // First time, retrieve events from the last 10 min
+                lastRun = {time: now - 10 * 60 * 1000};
+            }
+            setLastRun({time: now});
```

Comment:
You should set last run, only if fetch was successful
Else you may lose timeframes
If you intend to have it this way, then ignore